### PR TITLE
Added getById endpoint, that uses cache

### PR DESCRIPTION
**Pull request by danielbdias**
This PR adds another endpoint, that has a cache hit/miss example.

Example of get by Id with cache miss:
image

Example of get by Id with cache hit:
image

## Changes

- Adding getById endpoint with caching
- Update delete endpoint to invalidate cache
- Update cache duration

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: api/src/api.ts [hunk 1]
```diff
@@ -52,6 +54,10 @@ async function startApp() {
   ui.use(serve(resolve(__dirname, './ui')));
   app.use(mount('/', ui));
 
+  app.on('error', (err, ctx) => {
```
Comment (xoscar) [inline api/src/api.ts:57]: Do we need this?

File: api/src/handlers/getbyid.handler.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const getById = async (ctx, next) => {
+  const { id = '0' } = ctx.params || {};
+
+  const cachedPokemon = await cache.get(`pokemon-${id}`);
+  if (!!cachedPokemon) {
+    return cachedPokemon; // cache hit
+  }
+
+  const repository = getPokemonRepository();
```
Comment (xoscar) [inline api/src/handlers/getbyid.handler.ts:15]: This can be moved to like 6

File: api/src/handlers/remove.handler.ts [hunk 3]
```diff
@@ -11,6 +14,8 @@ const remove = async ctx => {
     return;
   }
 
+  cache.set(`pokemon-${id}`, null)
```
Comment (xoscar) [inline api/src/handlers/remove.handler.ts:17]: Shouldn't we remove the entry instead of setting it to `null`?

**Review by xoscar — CHANGES_REQUESTED** [review]
Hey @danielbdias GJ I left some comments

File: api/src/handlers/remove.handler.ts (hunk 3 above)
Comment (danielbdias) [reply api/src/handlers/remove.handler.ts:17]: It makes sense, I'll add these commands to our cache.

File: api/src/api.ts (hunk 1 above)
Comment (danielbdias) [reply api/src/api.ts:57]: It is more of a safeguard when something on our `docker compose` goes wrong silently. I used these logs to see why my connection with redis and rabbit was broken sometimes.

Comment (danielbdias) [reply api/src/api.ts]: Forget it, it did not help when I tried to inspect it again, so I'm removing it.

File: api/src/handlers/getbyid.handler.ts (hunk 2 above)
Comment (danielbdias) [reply api/src/handlers/getbyid.handler.ts]: Agreed and changed!

**Review by xoscar — APPROVED** [review]
GJ!

**Pull request merged**